pnet/Private-Networks-PSK-V1.md

```diff
@@ -1,50 +1,52 @@
+* Authenticity of data is not performed by the PSK Network and should be handled by a separate encryption suite such as secio or TLS1.3.
+
+### Enforcement
+* A node exposed to the environment variable `LIBP2P_FORCE_PNET=1` when started, must enforce the existence of a PSK Network compliant protector.
+  * If enforcement is turned on, and no protector is provided, the node must be terminated immediately.
+
+### Private Shared Key
+* A Private Shared Key is made available to the IPFS node, by adding a `swarm.key` file to the root of the IPFS Repo, prior to node creation.
+* The key file must comply to the format detailed at [Private Shared Key Format](#private-shared-key-format)
+
+#### Private Shared Key Format
+* The Private Shared Key file, `swarm.key`, must contain 3 lines.
+  * Line 1 must define the path-based multicodec for the key, `/key/swarm/psk/1.0.0/`.
+  * Line 2 must define the path-based multicodec in which the Private Key, on Line 3, is encoded, such as `/base16/`.
+  * Line 3 must contain the 32 byte Private Shared Key used for the private IPFS nodes, which must be encoded via the multicodec on Line 2.
+
+##### PSK File Example
+This example is for illustration purposes only. **DO NOT USE THIS KEY FOR PRODUCTION!!!**
+```txt
+/key/swarm/psk/1.0.0/
+/base16/
+0ff05394c733304ec26b3861ef23a4ffef9cd2e7d2040e028e5e9bb321d2eea3
+```
 
-### Cryptography of Private Networks
+## Cryptography of Private Networks (PNs)
 
 The cryptography behind PNs was chosen to have a minimal resource overhead but to maintain security guarantees that connections should be established with and only with nodes sharing the same PSK. We have decided to encrypt all traffic, thus reducing the possible attack surface of protocols that are part of IPFS/libp2p.
 
 It is important to mention that traffic in a private network is encrypted twice, once with PSK and once with the regular cryptographic stack for libp2p (secio or in the future TLS1.3). This allows the PSK layer to provide only above security guarantee, and for example not worrying about authenticity of the data. Possible replay attacks will be caught by the regular cryptographic layer above PNs layer.
 
-#### Choosing stream ciphers
+### Choosing stream ciphers
 
-We considered three stream siphers: AES-CTR, Salsa20 and ChaCha. Salsa20 and ChaCha are very similar ciphers, as ChaCha is fully based on Salsa20. And unfortunately, due of ChaCha's lack of adoption, we were not able to find vetted implementations in relevant programming languages. Because of this, the final consideration was between AES-CTR and Salsa20.
+We considered three stream ciphers: AES-CTR, Salsa20 and ChaCha. Salsa20 and ChaCha are very similar ciphers, as ChaCha is fully based on Salsa20. Unfortunately, due of ChaCha's lack of adoption, we were not able to find vetted implementations in relevant programming languages, which resulted in the ultimate consideration between AES-CTR and Salsa20.
```

Comment:
On the other hand, Rust has an AES and a ChaCha library but no Salsa20 library as far as I know.
However in my opinion a specification should decide which algorithm to use based on the virtues of the algorithm, not whether they are implemented in the relevant language.